client/src/logic/saved-decks/saved-decks.ts

```diff
@@ -94,3 +95,40 @@ export const convertLegacyDecks = (): number => {
 
 	return conversionCount
 }
+
+export const getCreatedTags: () => Array<Tag> = () => {
+	let lsKey
+	const tags = []
+
+	for (let i = 0; i < localStorage.length; i++) {
+		lsKey = localStorage.key(i)
+
+		if (lsKey?.includes('Tag_')) {
+			const key = localStorage.getItem(lsKey)
+			if (key) tags.push(JSON.parse(key) || {})
+		}
+	}
+	return tags.sort() as Array<Tag>
+}
+
+export const keysToTags = (tags: Array<string>): Array<Tag> => {
+	const savedTags = getCreatedTags()
+	const fullTags: Array<Tag> = []
+	tags.forEach((key) => {
+		const fullTag = savedTags.find((tag) => tag.key === key)
+		if (fullTag) fullTags.push(fullTag)
+	})
+	return fullTags
+}
+
+export const saveTag = (tag: Tag) => {
+	const createdTags = getCreatedTags()
+	const hash = 'Tag_' + tag.key
+	if (createdTags.find((createdTag) => createdTag.key === tag.key)) return
+	localStorage.setItem(hash, JSON.stringify(tag))
+}
+
+export const deleteTag = (tag: Tag) => {
```

Comment:
This does not remove references to the deleted tag in saved decks

Reply:
Fixed